x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/session/Configuration.java

```diff
@@ -21,12 +22,13 @@
     private final String clientId;
     private final boolean includeFrozenIndices;
     private final EqlSearchTask task;
+    private final TaskId taskId;
 
     @Nullable
     private QueryBuilder filter;
 
     public Configuration(String[] indices, ZoneId zi, String username, String clusterName, QueryBuilder filter, TimeValue requestTimeout,
-                         int size, boolean includeFrozen, String clientId, EqlSearchTask task) {
+                         int size, boolean includeFrozen, String clientId, String nodeId, EqlSearchTask task) {
```

Comment:
Since `nodeId` and `EqlSearchTask` are coupled, maybe they can be assembled into a compound object?

Reply:
Yes, I agree. The problem here that we really need TaskId (nodeID, id)  and we need a cancellation flag. The cancellation flag and id are coming from EqlSearchTask, but in order to assemble TaskId we need to also know the node. The EqlSearchTask is basically CancellableTasks and shouldn't really care about the node id. I [tried to clean it up a bit](https://github.com/elastic/elasticsearch/pull/54843/commits/00c14b034224d1424d18edc8ba49a37791fcce71) and removed dependency on EqlSearchTask for now. We will see how it goes. We might need to bring it back for progress updates, but I am not 100% sure that will be necessary.